### PR TITLE
Version bumps and fixes

### PR DESCRIPTION
**Pull request by JoseAlcerreca**
- Bumps to SDK 30 and latest stable deps
 - Removes mockito and adds navigation-testing
 - Cleanup

### CODE REVIEW

**Review by manuelvicnt — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -17,15 +18,16 @@ buildscript {
...
 
 plugins {
-    id "com.diffplug.gradle.spotless" version "3.24.0"
+    id "com.diffplug.spotless" version "5.12.5"
 }
 
 allprojects {
     repositories {
         google()
+        mavenCentral()
```
Comment (manuelvicnt) [inline build.gradle:27]: Is `mavenCentral()` needed?

File: app/src/sharedTest/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskFragmentTest.kt [hunk 2]
```diff
@@ -134,4 +117,17 @@ class AddEditTaskFragmentTest {
         assertEquals(tasks[0].title, "title")
         assertEquals(tasks[0].description, "description")
     }
+
+    private fun launchFragment(navController: TestNavHostController) {
+        val bundle = AddEditTaskFragmentArgs(
+            null,
+            getApplicationContext<Context>().getString(R.string.add_task)
```
Comment (manuelvicnt) [inline app/src/sharedTest/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskFragmentTest.kt:124]: Is `getApplicationContext` safe to use?

Comment (JoseAlcerreca) [reply app/src/sharedTest/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskFragmentTest.kt:124]: Yup

File: build.gradle (hunk 1 above)
Comment (JoseAlcerreca) [reply build.gradle:27]: removed

**Review by APinfinite — COMMENTED** [review]
app/build.gradle

File: app/build.gradle [hunk 3]
```diff
@@ -89,6 +89,11 @@ android {
     kotlinOptions {
         jvmTarget = "1.8"
     }
+
+    packagingOptions {
+        exclude 'META-INF/AL2.0'
+        exclude 'META-INF/LGPL2.1'
```
Comment (DoctorReyhanrizAbbas) [inline app/build.gradle:95]: Company.Online

**Pull request merged**